### PR TITLE
SONARIAC-546 Modify ExposeInstruction implementation to use Argument

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

**Comment by rudy-regazzoni-sonarsource** [comment]
First babystep, I'm tempted to move the static class `Port` in a separate file and have a proper `PortTest` to test it.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/ExposePortCheck.java [hunk 1]
```diff
@@ -87,14 +85,44 @@ private static boolean isBetween(int value, int min, int max) {
...
+      }
+    }
+
+    private void parseProtocol(String[] splittedProtocol) {
+      if (splittedProtocol.length > 1 && splittedProtocol[1].equalsIgnoreCase("udp")) {
+        protocol = Protocol.UDP;
+      } else {
+        protocol = Protocol.TCP;
+      }
     }
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/ExposePortCheck.java:125]: IMO A builder pattern would look nicer https://refactoring.guru/design-patterns/builder

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/ExposeInstruction.java [hunk 2]
```diff
@@ -22,5 +22,5 @@
 import java.util.List;
 
 public interface ExposeInstruction extends Instruction {
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/ExposeInstruction.java:24]: Rebase first to use the new interface

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM! Please rebase first and maybe change the construction of the port class to a builder pattern.

**Pull request merged**